data_model/src/transaction.rs

```diff
@@ -306,125 +142,56 @@ pub mod model {
         pub max_wasm_size_bytes: u64,
     }
 
-    /// Structure that represents the initial state of a transaction before the transaction receives any signatures.
-    #[derive(Debug, Clone, PartialEq, Eq)]
-    #[repr(transparent)]
-    pub struct TransactionBuilder {
-        /// [`Transaction`] payload.
-        pub payload: TransactionPayload,
-    }
-
-    /// Structure that represents the second state of the transaction after receiving at least one signature.
+    /// Transaction that contains at least one signature
     ///
-    /// `Iroha` and its clients use [`Transaction`] to send transactions over the network.
+    /// `Iroha` and its clients use [`Self`] to send transactions over the network.
     /// After a transaction is signed and before it can be processed any further,
     /// the transaction must be accepted by the `Iroha` peer.
     /// The peer verifies the signatures and checks the limits.
     #[version(n = 1, versioned = "VersionedSignedTransaction")]
     #[derive(
-        Debug,
-        Display,
-        Clone,
-        PartialEq,
-        Eq,
-        Hash,
-        Decode,
-        Encode,
-        Deserialize,
-        Serialize,
-        IntoSchema,
+        Debug, Display, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Serialize, IntoSchema,
     )]
-    #[display(fmt = "{self:?}")] // TODO ?
+    #[cfg_attr(not(feature = "std"), display(fmt = "Signed transaction"))]
+    #[cfg_attr(feature = "std", display(fmt = "{}", "self.hash()"))]
     #[ffi_type]
+    // TODO: All fields in this struct should be private
```

Comment:
Because we don't want them to be modified independently?

Reply:
Because we want the codebase to be version independent. Nobody should be using V1 structs outside data_model. All operations should be done on VersionedXXX structs